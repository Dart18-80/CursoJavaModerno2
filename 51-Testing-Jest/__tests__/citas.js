import Citas from '../js/classes/Citas.js';



describe('Probar la clase de Citas', () => {

    const citas = new Citas();

    const id = Date.now()

    test('Agregar una nueva cita', () => {
        const citaObj = {
            mascota: 'Kiba',
            propietario: 'Diego',
            telefono: '19030913',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citaObj.id = id;

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Akamaru',
            propietario: 'Diego',
            telefono: '19030913',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {

        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});