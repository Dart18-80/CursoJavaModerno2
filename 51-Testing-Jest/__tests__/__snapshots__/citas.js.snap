// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1639493201012,
      "mascota": "Akamaru",
      "propietario": "Diego",
      "sintomas": "Solo duerme",
      "telefono": "19030913",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1639493201012,
      "mascota": "Kiba",
      "propietario": "Diego",
      "sintomas": "Solo duerme",
      "telefono": "19030913",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
